### PR TITLE
fix: Add ramp view to assets page if there are no assets

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Adds a new screen to the assets page with the Safe QR Code and a link for on-ramping.

- [ ] Add correct on ramp safe app link

## Screenshots
Screenshot 2023-11-09 at 09 40 18

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/balances/AssetsTable/index.tsx [hunk 1]
```diff
@@ -247,11 +259,55 @@ const AssetsTable = ({
...
+            Add funds directly from your bank account or copy your address to send tokens from a different account.
+          </Typography>
+          <Box bgcolor="background.main" p={2} borderRadius="6px" alignSelf="flex-start" fontSize="14px">
+            <EthHashInfo address={safeAddress} shortAddress={false} showCopyButton hasExplorer avatarSize={24} />
+          </Box>
+          <Box alignSelf="flex-start">
+            {/* TODO: Insert link for Ramp app */}
+            <Link href={{ pathname: AppRoutes.apps.index, query: router.query }} passHref>
+              <Button variant="contained" size="small" startIcon={<AddIcon />}>
+                Buy Crypto
```
Comment (katspaugh) [inline src/components/balances/AssetsTable/index.tsx:303]:
```suggestion
                Buy crypto
```

🦸🏻‍♂️🐕

File: src/components/balances/AssetsTable/index.tsx [hunk 2]
```diff
@@ -137,6 +147,8 @@ const AssetsTable = ({
     [hiddenAssets, balances.items, showHiddenAssets],
   )
 
+  const hasNoAssets = balances.items.length === 1 && balances.items[0].balance === '0'
```
Comment (usame-algan) [inline src/components/balances/AssetsTable/index.tsx:150]: Not sure if there is a better way to find this out. Any ideas?

Comment (schmanu) [reply src/components/balances/AssetsTable/index.tsx:150]: I think this is the only way as our endpoint always returns the native token balance. Even if it is 0.

File: src/components/balances/AssetsTable/index.tsx [hunk 3]
```diff
@@ -247,11 +259,55 @@ const AssetsTable = ({
...
+            Add funds to get started
+          </Typography>
+          <Typography>
+            Add funds directly from your bank account or copy your address to send tokens from a different account.
+          </Typography>
+          <Box bgcolor="background.main" p={2} borderRadius="6px" alignSelf="flex-start" fontSize="14px">
+            <EthHashInfo address={safeAddress} shortAddress={false} showCopyButton hasExplorer avatarSize={24} />
+          </Box>
+          <Box alignSelf="flex-start">
+            {/* TODO: Insert link for Ramp app */}
```
Comment (schmanu) [inline src/components/balances/AssetsTable/index.tsx:300]: In order to find the app we should probably add a tag "on-ramping" to the Safe App so we can search for the appUrl by tag similar to the Safe Governance App links in the header.

**Review by schmanu — COMMENTED** [review]
Looks good. Before merging we have to adjust it with the link to the Ramp Safe app or if it doesn't exist on the current chain not display the Button.

**Comment by francovenica** [comment]
It works.

The message looks fine, 
the QR code works, 
the address is fine with icon and buttons to copy / block explorer
The "Buy" button only shows if the app is enabled in the network 
The message is gone if you receive tokens and shows up again if you empty your safe of tokens
The message shows if you have only "non-default" tokens and dissapears if you select "All tokens" from the "Token list" dropdown, which I consider the proper behaviour

Question:
Would it be possible to remove the "hide tokens" menu while that message shows up. It makes no sense to show if if there is no table to filter. 
I don't consider this an issue,  is a suggestion

Note:
I enabled Ramp app in sepolia for a minute in django to see that the "Buy" button shows up, which it did, but after disabling the app the button kept showing up. Maybe cache? I'll check tomorrow if it dissapears

**Comment by usame-algan** [comment]
Thanks for testing!

> I enabled Ramp app in sepolia for a minute in django to see that the "Buy" button shows up, which it did, but after disabling the app the button kept showing up. Maybe cache? I'll check tomorrow if it dissapears

I had the same issue when we enabled it on goerli but it disappeared after a while. Definitely a caching issue.

**Pull request merged**